wizard/vote_on_onchain_governance.py

```diff
@@ -0,0 +1,208 @@
+    try:
+        _do_main(cli_args)
+    except errors.KnownError as err:
+        ux.show_critical_error(traceback.format_exc())
+        ux.show_critical_error(err.get_pretty())
+        return 1
+
+def _do_main(cli_args: List[str]):
+    parser = ArgumentParser()
+    parser.add_argument("--network", choices=CONFIGURATIONS.keys(), required=True, help="network name")
+    parser.add_argument("--wallets", required=True, help="path to wallets configuration file")
+    parser.add_argument("--auth", required=True, help="auth registration file")
+    parser.add_argument("--gas-price", type=int, default=DEFAULT_GAS_PRICE, help="min gas price")
+    sub = parser.add_subparsers(dest="action", required=True)
+
+    vote_p = sub.add_parser("vote", help="cast votes (bulk) for a proposal")
+    vote_p.add_argument("--proposal", type=int, required=True, help="proposal nonce / id")
+    vote_p.add_argument("--choice", choices=["yes", "no", "abstain"], required=True, help="vote choice")
+
+    close_p = sub.add_parser("close", help="close a proposal")
+    close_p.add_argument("--proposal", type=int, required=True, help="proposal nonce / id")
+
+    propose_p = sub.add_parser("propose", help="create a new proposal")
+    propose_p.add_argument("--commit-hash", required=True, help="commit hash for proposal content")
+    propose_p.add_argument("--start-epoch", type=int, required=True, help="start vote epoch")
+    propose_p.add_argument("--end-epoch", type=int, required=True, help="end vote epoch")
+    propose_p.add_argument("--fee-egld", type=float, default=0.0, help="native EGLD amount to attach")
+
+    clear_p = sub.add_parser("clear", help="clear ended proposals for given proposers")
+    clear_p.add_argument("--proposers", nargs="+", required=True, help="list of bech32 proposer addresses")
+
+    claim_p = sub.add_parser("claim-fees", help="claim accumulated governance fees")
+
+    chgcfg_p = sub.add_parser("change-config", help="change governance configuration")
+    chgcfg_p.add_argument("--proposal-fee", type=int, required=True, help="proposal fee (wei)")
+    chgcfg_p.add_argument("--lost-proposal-fee", type=int, required=True, help="lost proposal fee (wei)")
+    chgcfg_p.add_argument("--min-quorum", type=int, required=True, help="minimum quorum")
+    chgcfg_p.add_argument("--min-veto-threshold", type=int, required=True, help="minimum veto threshold")
+    chgcfg_p.add_argument("--min-pass-threshold", type=int, required=True, help="minimum pass threshold")
+
+    args = parser.parse_args(cli_args)
+
+    network = args.network
+    configuration = CONFIGURATIONS[network]
+    entrypoint = MyEntrypoint(configuration)
+
+    proxy_url = getattr(configuration, "proxy", None) or getattr(configuration, "proxy_url", None)
+    if not proxy_url:
+        raise errors.KnownError("CONFIGURATION is missing 'proxy' URL.")
+    provider = ProxyNetworkProvider(proxy_url)
+
+    chain_id = getattr(configuration, "chain_id", None)
+    if not chain_id:
```

Comment:
not sure the checks for proxy and chain id are necessary, but it's ok nevertheless.